aws/resource_aws_servicecatalog_product.go

```diff
@@ -0,0 +1,447 @@
+		"name",
+		"owner",
+		"support_description",
+		"support_email",
+		"support_url",
+	) {
+		input := &servicecatalog.UpdateProductInput{
+			Id: aws.String(d.Id()),
+		}
+
+		if v, ok := d.GetOk("accept_language"); ok {
+			input.AcceptLanguage = aws.String(v.(string))
+		}
+
+		if v, ok := d.GetOk("description"); ok {
+			input.Description = aws.String(v.(string))
+		}
+
+		if v, ok := d.GetOk("distributor"); ok {
+			input.Distributor = aws.String(v.(string))
+		}
+
+		if v, ok := d.GetOk("name"); ok {
+			input.Name = aws.String(v.(string))
+		}
+
+		if v, ok := d.GetOk("owner"); ok {
+			input.Owner = aws.String(v.(string))
+		}
+
+		if v, ok := d.GetOk("support_description"); ok {
+			input.SupportDescription = aws.String(v.(string))
+		}
+
+		if v, ok := d.GetOk("support_email"); ok {
+			input.SupportEmail = aws.String(v.(string))
+		}
+
+		if v, ok := d.GetOk("support_url"); ok {
+			input.SupportUrl = aws.String(v.(string))
+		}
+
+		err := resource.Retry(iamwaiter.PropagationTimeout, func() *resource.RetryError {
+			_, err := conn.UpdateProduct(input)
+
+			if tfawserr.ErrMessageContains(err, servicecatalog.ErrCodeInvalidParametersException, "profile does not exist") {
+				return resource.RetryableError(err)
+			}
+
+			if err != nil {
+				return resource.NonRetryableError(err)
+			}
+
+			return nil
+		})
+
+		if tfresource.TimedOut(err) {
+			_, err = conn.UpdateProduct(input)
+		}
+
+		if err != nil {
+			return fmt.Errorf("error updating Service Catalog Product (%s): %w", d.Id(), err)
+		}
+	}
+
+	if d.HasChange("tags_all") {
+		o, n := d.GetChange("tags_all")
+
+		if err := keyvaluetags.ServiceCatalogProductUpdateTags(conn, d.Id(), o, n); err != nil {
+			return fmt.Errorf("error updating tags for Service Catalog Product (%s): %w", d.Id(), err)
+		}
+	}
+
+	return resourceAwsServiceCatalogProductRead(d, meta)
+}
+
+func resourceAwsServiceCatalogProductDelete(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).scconn
+
+	input := &servicecatalog.DeleteProductInput{
+		Id: aws.String(d.Id()),
+	}
+
+	if v, ok := d.GetOk("accept_language"); ok {
+		input.AcceptLanguage = aws.String(v.(string))
+	}
+
+	_, err := conn.DeleteProduct(input)
+
+	if tfawserr.ErrCodeEquals(err, servicecatalog.ErrCodeResourceNotFoundException) {
+		return nil
+	}
+
+	if err != nil {
+		return fmt.Errorf("error deleting Service Catalog Product (%s): %w", d.Id(), err)
+	}
+
+	if _, err := waiter.ProductDeleted(conn, d.Get("accept_language").(string), d.Id()); err != nil {
+		return fmt.Errorf("error waiting for product (%s) to be deleted: %w", d.Id(), err)
+	}
+
+	return nil
+}
+
+const (
+	ServiceCatalogAcceptLanguageEnglish  = "en"
+	ServiceCatalogAcceptLanguageJapanese = "jp"
+	ServiceCatalogAcceptLanguageChinese  = "zh"
+)
+
+func AcceptLanguage_Values() []string {
+	return []string{
+		ServiceCatalogAcceptLanguageEnglish,
+		ServiceCatalogAcceptLanguageJapanese,
+		ServiceCatalogAcceptLanguageChinese,
+	}
+}
```

Comment:
Maybe move to  a new file `aws/internal/service/servicecatalog/consts.go`?

Reply:
The only other example I found of this was in route53 with `enum.go` so I went that direction.